ui/src/widgets/ui-button/UIButton.vue

```diff
@@ -93,26 +81,23 @@ export default {
                 return
             }
             if (typeof updates.label !== 'undefined') {
-                this.dynamic.label = updates.label
+                this.setDynamicProperties({ label: updates.label })
             }
             if (typeof updates.icon !== 'undefined') {
-                this.dynamic.icon = updates.icon
+                this.setDynamicProperties({ icon: updates.icon })
             }
             if (typeof updates.iconPosition !== 'undefined') {
-                this.dynamic.iconPosition = updates.iconPosition
+                this.setDynamicProperties({ iconPosition: updates.iconPosition })
             }
             if (typeof updates.buttonColor !== 'undefined') {
-                this.dynamic.buttonColor = updates.buttonColor
+                this.setDynamicProperties({ buttonColor: updates.buttonColor })
             }
             if (typeof updates.textColor !== 'undefined') {
-                this.dynamic.textColor = updates.textColor
+                this.setDynamicProperties({ textColor: updates.textColor })
             }
             if (typeof updates.iconColor !== 'undefined') {
-                this.dynamic.iconColor = updates.iconColor
+                this.setDynamicProperties({ iconColor: updates.iconColor })
             }
```

Comment:
Joe, I think we can both cement and simplify this behaviour with an additional helper. 

### Additional helper, with guards and the `undefined` pattern built in for 100% consistency

```
                updateDynamicProperty(prop, config) {
                    if (!prop || !config || typeof config !== 'object') {
                       return // maybe better to throw error here as this is likely program error
                    }
                    if (typeof config[prop] === 'undefined') {
                       return // no change
                    }
                    this.$store.commit('ui/widgetState', {
                        widgetId: this.id,
                        config
                    })
                },
```

### Now, all calls, by all nodes (contrib too) become consistent
```
          this.updateDynamicProperty('label', updates)
          this.updateDynamicProperty('icon', updates)
          this.updateDynamicProperty('iconPosition', updates)
          this.updateDynamicProperty('buttonColor', updates)
          this.updateDynamicProperty('textColor', updates)
          this.updateDynamicProperty('iconColor', updates)
```


---

If necessary (i.e there is a reason to permit `undefined` as a value) we can add an options object to the `updateDynamicProperty` e.g.  `this.updateDynamicProperty('label', updates, { allowUndefined: true })`

I realise this is a little extra work but as this is a large change already, we have the opportunity to cement good practice at the core.

Happy to discuss over a call.